tests/support/kubernetes.py

```diff
@@ -580,6 +755,177 @@ async def readline() -> bytes:
+            Namespace in which to create the object.
+        body
+            Object to create.
+
+        Raises
+        ------
+        ApiException
+            Raised with 409 status if the object already exists.
+
+        """
+        self._maybe_error("create_namespaced_ingress", namespace, body)
+        self._update_metadata(
+            body, "networking.k8s.io/v1", "Ingress", namespace
+        )
+        name = body.metadata.name
+        body.status = V1IngressStatus(
+            load_balancer=V1LoadBalancerStatus(
+                ingress=[
+                    V1LoadBalancerIngress(ip="127.0.0.1"),
+                ]
+            )
+        )
+        self._store_object(namespace, "Ingress", name, body)
+
+    async def delete_namespaced_ingress(
+        self, name: str, namespace: str
+    ) -> V1Status:
+        """Delete an ingress object.
+
+        Parameters
+        ----------
+        name
+            Name of ingress to delete.
+        namespace
+            Namespace of ingress to delete.
+
+        Returns
+        -------
+        V1Status
+            Success status.
+
+        Raises
+        ------
+        ApiException
+            Raised with 404 status if the pod was not found.
+        """
+        self._maybe_error("delete_namespaced_ingress", name, namespace)
+        return self._delete_object(namespace, "Ingress", name)
+
+    async def read_namespaced_ingress(
+        self, name: str, namespace: str
+    ) -> V1Ingress:
+        """Read a ingress object.
+
+        Parameters
+        ----------
+        name
+            Name of the ingress.
+        namespace
+            Namespace of the ingress.
+
+        Returns
+        -------
+        V1Ingress
+            Ingress object.
+
+        Raises
+        ------
+        ApiException
+            Raised with 404 status if the ingress was not found.
+        """
+        self._maybe_error("read_namespaced_ingress", name, namespace)
+        return self._get_object(namespace, "Ingress", name)
+
+    # JOB API
+
+    async def create_namespaced_job(self, namespace: str, body: V1Job) -> None:
+        """Create a job object.  This will in turn create a pod object.  That
+        pod object will have its status set to "Running", and then the Job
+        status will have ``active`` set to 1.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to create the object.
+        body
+            Object to create.
+
+        Raises
+        ------
+        ApiException
+            Raised with 409 status if the object already exists.
+
+        """
+        self._maybe_error("create_namespaced_job", namespace, body)
+        name = body.metadata.name
+        await self.create_namespaced_pod(
+            namespace,
+            V1Pod(
+                metadata=V1ObjectMeta(name=name),
+            ),
+        )
+        pod = await self.read_namespaced_pod(name, namespace)
+        pod.status.phase = "Running"
```

Comment:
Don't do this -- the newly created pod should honor the default pod phase setting.

Reply:
done